managed/src/main/java/com/yugabyte/yw/controllers/AccessKeyController.java

```diff
@@ -98,88 +81,71 @@ public Result create(UUID customerUUID, UUID providerUUID) {
+      Http.MultipartFormData.FilePart filePart = multiPartBody.getFile("keyFile");
+      File uploadedFile = (File) filePart.getFile();
+      if (keyType == null || uploadedFile == null) {
+        return ApiResponse.error(BAD_REQUEST, "keyType and keyFile params required.");
       }
-
-      // In case of onprem provider, we add a couple of additional attributes like passwordlessSudo
-      // and create a preprovision script
-      if (region.provider.code.equals(onprem.name())) {
-        templateManager.createProvisionTemplate(
-          accessKey,
-          airGapInstall,
-          formData.get().passwordlessSudoAccess,
-          formData.get().installNodeExporter,
-          formData.get().nodeExporterPort,
-          formData.get().nodeExporterUser
-        );
+      accessKey = accessManager.uploadKeyFile(
+          region.uuid, uploadedFile, keyCode, keyType, sshUser, sshPort, airGapInstall,
+          skipProvisioning);
+    } else if (keyContent != null && !keyContent.isEmpty()) {
+      if (keyType == null) {
+        return ApiResponse.error(BAD_REQUEST, "keyType params required.");
       }
-    } catch(RuntimeException | IOException e) {
-      LOG.error(e.getMessage());
-      return ApiResponse.error(INTERNAL_SERVER_ERROR, "Unable to create access key: " + keyCode);
+      // Create temp file and fill with content
+      Path tempFile = Files.createTempFile(keyCode, keyType.getExtension());
+      Files.write(tempFile, keyContent.getBytes());
+
+      // Upload temp file to create the access key and return success/failure
+      accessKey = accessManager.uploadKeyFile(
+          regionUUID, tempFile.toFile(), keyCode, keyType, sshUser, sshPort, airGapInstall,
+          skipProvisioning);
+    } else {
+      accessKey = accessManager.addKey(
+          regionUUID, keyCode, sshPort, airGapInstall, skipProvisioning);
+    }
+
+    // In case of onprem provider, we add a couple of additional attributes like passwordlessSudo
+    // and create a preprovision script
+    if (region.provider.code.equals(onprem.name())) {
+      templateManager.createProvisionTemplate(
+        accessKey,
+        airGapInstall,
+        formData.get().passwordlessSudoAccess,
+        formData.get().installNodeExporter,
+        formData.get().nodeExporterPort,
+        formData.get().nodeExporterUser
+      );
     }
     Audit.createAuditEntry(ctx(), request(), Json.toJson(formData.data()));
     return ApiResponse.success(accessKey);
   }
 
   public Result delete(UUID customerUUID, UUID providerUUID, String keyCode) {
-    String validationError = validateUUIDs(customerUUID, providerUUID);
-    if (validationError != null) {
-      return ApiResponse.error(BAD_REQUEST, validationError);
-    }
-
-    AccessKey accessKey = AccessKey.get(providerUUID, keyCode);
-    if (accessKey == null) {
-      return ApiResponse.error(BAD_REQUEST, "KeyCode not found: " + keyCode);
-    }
-
+    validateUUIDs(customerUUID, providerUUID);
```

Comment:
I think 
```
    Customer customer = Customer.getOrBadRequest(customerUUID);
    Universe universe = Universe.getValidUniverseOrBadRequest(universeUUID, customer);
``` should do what validateUUID is doing.

Reply:
yes, checking customer plus whether the provider is linked to that customer or not.